File: go/raindrops/raindrops.go
package raindrops

import (
	"fmt"
	"strconv"
)

func Convert(input int) (result string) {
	drops := []struct {
		Value int
		Key   string
	}{
		{3, "Pling"},
		{5, "Plang"},
		{7, "Plong"},
	}

	for _, drop := range drops {
		if (input % drop.Value) == 0 {
			result = fmt.Sprintf("%s%s", result, drop.Key)
		}
	}

	if result == "" {
		result = strconv.Itoa(input)
	}

	return result
}

File: nim/raindrops/raindrops.nim
import strutils
import strformat
import sugar

func convert*(input: int): string =
    const
        factors = [
            (num: 3, value: "Pling"),
            (num: 5, value: "Plang"),
            (num: 7, value: "Plong"),
        ]

    let data = collect(newSeq):
        for factor in factors:
            if input mod factor.num == 0:
                factor.value

    result = if data.len > 0: data.join("") else: $input
